pkg/scheduler/apis/config/types.go

```diff
@@ -0,0 +1,339 @@
+)
+
+const (
+	// SchedulerDefaultLockObjectNamespace defines default scheduler lock object namespace ("kube-system")
+	SchedulerDefaultLockObjectNamespace string = metav1.NamespaceSystem
+
+	// SchedulerDefaultLockObjectName defines default scheduler lock object name ("kube-scheduler")
+	SchedulerDefaultLockObjectName = "kube-scheduler"
+
+	// SchedulerPolicyConfigMapKey defines the key of the element in the
+	// scheduler's policy ConfigMap that contains scheduler's policy config.
+	SchedulerPolicyConfigMapKey = "policy.cfg"
+
+	// SchedulerDefaultProviderName defines the default provider names
+	SchedulerDefaultProviderName = "DefaultProvider"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// KubeSchedulerConfiguration configures a scheduler
+type KubeSchedulerConfiguration struct {
+	metav1.TypeMeta
+
+	// AlgorithmSource specifies the scheduler algorithm source.
+	// TODO(#87526): Remove AlgorithmSource from this package
+	// DEPRECATED: AlgorithmSource is removed in the v1alpha2 ComponentConfig
+	AlgorithmSource SchedulerAlgorithmSource
+
+	// LeaderElection defines the configuration of leader election client.
+	LeaderElection KubeSchedulerLeaderElectionConfiguration
+
+	// ClientConnection specifies the kubeconfig file and client connection
+	// settings for the proxy server to use when communicating with the apiserver.
+	ClientConnection componentbaseconfig.ClientConnectionConfiguration
+	// HealthzBindAddress is the IP address and port for the health check server to serve on,
+	// defaulting to 0.0.0.0:10251
+	HealthzBindAddress string
+	// MetricsBindAddress is the IP address and port for the metrics server to
+	// serve on, defaulting to 0.0.0.0:10251.
+	MetricsBindAddress string
+
+	// DebuggingConfiguration holds configuration for Debugging related features
+	// TODO: We might wanna make this a substruct like Debugging componentbaseconfig.DebuggingConfiguration
+	componentbaseconfig.DebuggingConfiguration
+
+	// DisablePreemption disables the pod preemption feature.
+	DisablePreemption bool
+
+	// PercentageOfNodeToScore is the percentage of all nodes that once found feasible
+	// for running a pod, the scheduler stops its search for more feasible nodes in
+	// the cluster. This helps improve scheduler's performance. Scheduler always tries to find
+	// at least "minFeasibleNodesToFind" feasible nodes no matter what the value of this flag is.
+	// Example: if the cluster size is 500 nodes and the value of this flag is 30,
+	// then scheduler stops finding further feasible nodes once it finds 150 feasible ones.
+	// When the value is 0, default percentage (5%--50% based on the size of the cluster) of the
+	// nodes will be scored.
+	PercentageOfNodesToScore int32
+
+	// Duration to wait for a binding operation to complete before timing out
+	// Value must be non-negative integer. The value zero indicates no waiting.
+	// If this value is nil, the default value will be used.
+	BindTimeoutSeconds int64
```

Comment:
comment and declaration don't agree - is it int64 or *int64?

Reply:
Interesting. This is from community. ./staging/src/k8s.io/kube-scheduler/config/v1alpha2/types.go has the field defined as *int64. Looks like other fields in KubeSchedulerConfiguration struct have similar usage. It seems KubeSchedulerConfiguration  struct defined in pkg/scheduler/apis/config/types.go is used to get argument from commandline. generated conversation handles the int64 to pointer 64 automatically. The comment seems a copied over and a tradition for those fields. I think it is ok to keep it as it is.